### PR TITLE
rill init-example with new cli

### PR DESCRIPTION
**Pull request by rakeshsharma14317**

### CODE REVIEW

**Review by begelundmuller — CHANGES_REQUESTED** [review]
1. Fix failing CI
2. Put the example embed logic in `cli/pkg/examples/examples.go` and put the embedded files in `cli/pkg/examples/embed/dist/` using the same approach as for the SPA. The gitignore probably needs to be `/cli/pkg/examples/embed/*/*` or something like that

File: examples/default/sources/e_shop_clothing.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+type: "file"
+path: "data/e-shop-clothing.csv"
```
Comment (begelundmuller) [inline examples/default/sources/e_shop_clothing.yaml:2]: Use:

```
type: "https"
uri: https://cdn.rilldata.com/examples/0.16/ecomm-click-stream/e-shop-clothing.csv
```

File: examples/default/sources/global_landslide_catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+type: "file"
+path: "data/global-landslide-catalog.csv"
```
Comment (begelundmuller) [inline examples/default/sources/global_landslide_catalog.yaml:2]: Use:

```
type: "https"
uri: https://cdn.rilldata.com/examples/0.16/global-landslide-catalog/global-landslide-catalog.csv
```

File: examples/default/sources/iot_telemetry_data.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+type: "file"
+path: "data/iot-telemetry-data.csv"
```
Comment (begelundmuller) [inline examples/default/sources/iot_telemetry_data.yaml:2]: Use:
```
type: "https"
uri: https://cdn.rilldata.com/examples/0.16/iot-env-sensor/iot-telemetry-data.csv
```

File: cli/cmd/initialize/init.go [hunk 4]
```diff
@@ -3,22 +3,40 @@ package initialize
...
 	"fmt"
 
+	example "github.com/rilldata/rill/cli/pkg/examples"
 	"github.com/spf13/cobra"
 )
 
 // InitCmd represents the init command
 func InitCmd() *cobra.Command {
-	var exampleProject string
+	var projectName string
```
Comment (begelundmuller) [inline cli/cmd/initialize/init.go:12]: Call this `exampleName` to prevent confusion

File: cli/pkg/examples/example.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+func getExampleProject(projectName string) (string, error) {
+	examplesPath := "embed/dist/" + projectName
+	_, err := exampleFS.ReadDir(examplesPath)
+	if err != nil {
+		return "", err
+	}
+	return examplesPath, nil
+}
+
+func CopyDir(origin string, dst string) (err error) {
```
Comment (begelundmuller) [inline cli/pkg/examples/example.go:44]: Make it `copyDir` since it's an internal function

File: cli/pkg/examples/example.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	"embed"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+//go:embed embed
+var exampleFS embed.FS
+
+func ListExamples() ([]string, error) {
```
Comment (begelundmuller) [inline cli/pkg/examples/example.go:13]: Just make it `List`, it's already called as `example.List()`. Same for `InitExample`, just call it `Init`

**Review by begelundmuller — COMMENTED** [review]
Added notes for remote sources, so you can remove the `examples/default/data` folder and save some space :)

File: examples/default/sources/e_shop_clothing.yaml (hunk 1 above)
Comment (rakeshsharma14317) [reply examples/default/sources/e_shop_clothing.yaml:2]: Yeah that make sense we better not save data here, we can more examples in future with http. removing data and addressing these changes

File: .gitignore [hunk 7]
```diff
@@ -19,14 +19,14 @@ export/
...
 /rill
+/cli/pkg/examples/embed/dist
 
 # data files
 *.duckdb
 *.duckdb.wal
 *.db
 *.wal
 *.parquet
-*.csv
```
Comment (AdityaHegde) [inline .gitignore:29]: We used to have some random csvs on the root. So better to keep it and force add the ones we need to add.

File: Makefile [hunk 8]
```diff
@@ -4,6 +4,8 @@ cli:
 	npm run build
 	mkdir -p cli/pkg/web/embed/dist
 	cp -r web-local/build/ cli/pkg/web/embed/dist
+	mkdir -p cli/pkg/examples/embed/dist
```
Comment (AdityaHegde) [inline Makefile:7]: Add a cleanup step to remove rogue files from older build

File: cli/README.md [hunk 9]
```diff
@@ -9,5 +9,28 @@
...
+2. To install it via homebrew
+```
+brew install rilldata/rill-developer/rill
+rill --help
+```
+
+3. You can also try our example:
+    1. List of available examples
+    ```
+    rill init list
```
Comment (AdityaHegde) [inline cli/README.md:21]: Looking at this i dont feel it makes sense to list available examples on `rill init list`. Should there perhaps be `rill init --example list` or something like that?

@hamilton @magorlick Any comments?

File: cli/README.md [hunk 10]
```diff
@@ -9,5 +9,28 @@
...
+    ```
+    2. Initialize the example project, default to `default`, default directory `.`
+    ```
+    rill init --example <example project name> --dir <directory to migrate example project>
```
Comment (AdityaHegde) [inline cli/README.md:25]: Is this upto date?

File: .gitignore (hunk 7 above)
Comment (rakeshsharma14317) [reply .gitignore:29]: Yeah make sense, Removed as we were pushing the data files as well, reverted back

File: cli/README.md (hunk 9 above)
Comment (rakeshsharma14317) [reply cli/README.md:21]: yeah just added for now, we can update when required. left for the product to test and give any suggestion on this part

Comment (rakeshsharma14317) [reply cli/README.md:21]: @AdityaHegde do you mean we should example as subcommand?

File: cli/README.md (hunk 10 above)
Comment (rakeshsharma14317) [reply cli/README.md:25]: If you mean it this complete, then no, Just added the some of commands which are working as per new logic. Still CLI is in progress. For few commands logic is not implemented yet.

File: cli/README.md (hunk 9 above)
Comment (begelundmuller) [reply cli/README.md:21]: Updated it to:
```
# creates an empty project
rill init

# creates the default example project
rill init --example

# creates a specific example project
rill init --example NAME

# lists available example projects
rill init --list-examples
```

We can iterate on this during QA, this gets the functionality in there

**Pull request merged**